### ISSUE TITLE
Topbar won't hide automatically after entering fullscreen

### ISSUE THREAD

**Issue by hooke007** [tomasklaen/uosc#880]
Reproduced on Mac : If the cursor is near the topbar before cycling fullscreen...

**Comment by tomasklaen** [comment]
We already do `cursor:leave()` here:

https://github.com/tomasklaen/uosc/blob/6fa34c31d0a5290dee83282205768d15111df7d8/src/uosc/main.lua#L425-L430

Don't know what else we can do.

Or do you mean that the cursor is at the top of the screen, and when the mpv gets fullscreened, the top bar lands below it, then mpv sends us the new cursor position, so the top bar becomes visible?

**Comment by hooke007** [comment]
Yeah I do check the function there. It's odd that it doesn't work for Mac.

https://github.com/tomasklaen/uosc/assets/41094733/3d57b1f5-a43a-4d1e-990d-afbaa8bb2e09

**Comment by hooke007** [comment]
[5.0.0](https://github.com/tomasklaen/uosc/releases/tag/5.0.0) works well. I'll try to trace deeper.

**Comment by christoph-heinrich** [comment]
Is that new? There have been a lot of changes to the mac code in mpv lately.

**Comment by hooke007** [comment]
> There have been a lot of changes to the mac code in mpv lately.

Those commits doesn't have effect on this issue. Tested both with mpv0.36 --- master.

**Comment by christoph-heinrich** [comment]
Idk if this will help here, but maybe logging when which property changes helps in figuring out what's different on mac.

```lua
do
	local msg = require('mp.msg')
	local utils = require('mp.utils')
	do
		local orig_fn = mp.observe_property
		mp.observe_property = function(name, _type, callback)
			orig_fn(name, _type, function(pname, val)
				msg.info("[observe_property]", pname, type(val),
					type(val) == 'table' and utils.format_json(val) or val)
				callback(pname, val)
			end)
		end
	end
end
```

Putting this at the start of main.lua should end up logging all property change callbacks.

**Comment by hooke007** [comment]
[mac.txt](https://github.com/tomasklaen/uosc/files/14897981/mac.txt)

**Comment by AndydeCleyre** [comment]
> We already do `cursor:leave()` here:
> 
> https://github.com/tomasklaen/uosc/blob/6fa34c31d0a5290dee83282205768d15111df7d8/src/uosc/main.lua#L425-L430
> 

Does that get run in the case mpv starts in fullscreen? I often start videos without a mouse at hand and get stuck with the title.

**Comment by christoph-heinrich** [comment]
> > We already do `cursor:leave()` here:
> > https://github.com/tomasklaen/uosc/blob/6fa34c31d0a5290dee83282205768d15111df7d8/src/uosc/main.lua#L425-L430
> 
> Does that get run in the case mpv starts in fullscreen? I often start videos without a mouse at hand and get stuck with the title.

That runs every time the `fullscreen` or `window-maximized` properties change, and also once on initialization for both.

**Comment by AndydeCleyre** [comment]
I still am stuck with the title and controls displayed every time I launch mpv, until I get a mouse. It doesn't matter where the cursor is when I launch. I'm using Arch Linux.

**Comment by AndydeCleyre** [comment]
I haven't looked at/for debugging docs for uosc yet, but soon I will so I can at least confirm that's what's happening here.

**Comment by AndydeCleyre** [comment]
I added printing of x and y at the beginning of `function cursor:move(x, y)`. 

I don't know if this is expected or not, but when I start a video and quit, without moving the cursor at all, both are reported as `inf`. This is reported twice in a row early in the output, and twice in a row again late in the output.

**Comment by AndydeCleyre** [comment]
It appears that in `function TopBar:render()`, when I start mpv, the cursor is reported as `(0.5, 0.5)`. This isn't accurate, which my be intentional. But anyway I can fix my problem by adding an early return when the cursor reports at those coordinates:

```lua
if cursor.x == 0.5 and cursor.y == 0.5 then return end
```

So far, so good.

**Comment by tomasklaen** [comment]
We add `0.5` to cursor positions so that the cursor is reported to be in the middle of a pixel for more accurate timeline cursor position & seeking calculations. This means the raw cursor position was reported as (0,0), which shouldn't get past the `first_real_mouse_move_received` check.

Try adding this on the first line of `cursor:move(x, y)` function inside `lib/cursor.lua`:

```lua
print('cursor:', utils.to_string(x), utils.to_string(y))
```

so we can see what values does your mpv reports on startup before you move anything.

**Comment by tomasklaen** [comment]
Try this change:

```diff
--- a/src/uosc/lib/cursor.lua
+++ b/src/uosc/lib/cursor.lua
@@ -289,7 +301,7 @@ function cursor:move(x, y)
 	-- displays the top bar, so we hardcode cursor position as infinity until
 	-- we receive a first real mouse move event with coordinates other than 0,0.
 	if not self.first_real_mouse_move_received then
-		if x > 0 and y > 0 then
+		if x > 0 and y > 0 and x < 99999999 and y < 99999999 then
 			self.first_real_mouse_move_received = true
 		else
 			x, y = math.huge, math.huge
```

**Comment by AndydeCleyre** [comment]
Yes, thanks! With that, the top bar displays for just 1-2 seconds, then disappears, which is great.
